vmss-prototype/vmss-prototype

```diff
@@ -352,6 +352,98 @@ def get_vmss_images(subscription, resource_group, sig_name, image_definition):
+    be created for each new node.
+    The yaml specifically supports running on a cordoned node and it
+    also specifically calls out the node in question.
+    :param node_name:  The node, after it has been drained but it is still up
+    :returns: True if the tweaks were successfully deployed
+    """
+    namespace = os.getenv('POD_NAMESPACE', None)
+    if not namespace:
+        logging.info('No POD_NAMESPACE defined, can not do image tweaks')
+        return False
+
+    image = os.getenv('POD_IMAGE', None)
+    if not image:
+        logging.info('No POD_IMAGE defined, can not do image tweaks')
+        return False
+
+    pod_name = 'vmss-prototype-tweaks-{0}'.format(node_name)
+
+    # This is a pod yaml for our image tweak we will do on the target node
+    # that will be our new prototype image.
+    yaml_text = ("apiVersion: v1\n"
+                 "kind: Pod\n"
+                 "metadata:\n"
+                 "  name: '{0}'\n"
+                 "  namespace: '{1}'\n"
+                 "spec:\n"
+                 "  containers:\n"
+                 "  - command:\n"
+                 "    - /bin/cp\n"
+                 "    - /dev/null\n"
+                 "    - /hostetc/machine-id\n"
+                 "    image: '{2}'\n"
+                 "    imagePullPolicy: IfNotPresent\n"
+                 "    name: tweaks\n"
+                 "    volumeMounts:\n"
+                 "    - mountPath: /hostetc\n"
+                 "      name: etc\n"
+                 "      readOnly: false\n"
+                 "  hostNetwork: true\n"
+                 "  imagePullSecrets:\n"
+                 "  - name: skyman-acr\n"
+                 "  nodeSelector:\n"
+                 "    kubernetes.io/hostname: '{3}'\n"
+                 "  priorityClassName: skyman-cluster-daemonset\n"
+                 "  restartPolicy: Never\n"
+                 "  tolerations:\n"
+                 "  - effect: NoSchedule\n"
+                 "    operator: Exists\n"
+                 "  - effect: NoExecute\n"
+                 "    operator: Exists\n"
+                 "  volumes:\n"
+                 "  - hostPath:\n"
+                 "      path: /etc\n"
+                 "      type: Directory\n"
+                 "    name: etc\n"
+                ).format(pod_name, namespace, image, node_name)
+
+    run(['kubectl', 'delete', 'pod',
+         '--namespace', namespace,
+         pod_name
+         ], check=False)
+
+    _, _, rc = run(['kubectl', 'create', '-f', '-'], stdin=yaml_text, retries=3, check=False)
+
+    if rc:
+        logging.info('Could not successfully deploy the image tweaks')
+        return False
+
+    # We use this to watch for the pod to complete.  Unfortunately there is
+    # no condition for completed so the kubectl wait command can not work
+    # here and trying to use it for Ready ends up in a race with completion.
+    # Thus, we use the logs method.  Not as pretty but it works.
+    run(['kubectl', 'logs',
```

Comment:
would be cleaner with a job.

Reply:
Maybe it is worth it for this case to use a job and the overhead in kubernetes and let the job do all of this (similar) work when I ask to wait for the job to complete.  I am unsure it buys us much.

Reply:
For now, I would rather leave it like this.  We can look back on this later but jobs do not add much over a pod when I am really just needing to make a one-shot pod for a specific node that is about to go away.